### PR TITLE
feat(bo): [SFEQS-1879, SFEQS-1898, SFEQS-1900] add backoffice front-end pages

### PR DESCRIPTION
**Pull request by lucacavallaro**
#### List of Changes

Added the following pages, that allows the management of `api keys`
  - `/api-keys` 
  - `/api-keys/create`
  - `/api-keys/[id]`
 
These pages contain many components, including a form (`CreateApiKeyForm.tsx`) and custom input component (`EditableList`)

#### Screenshots (if appropriate):

image
image
image
image
image

### CODE REVIEW

File: apps/io-sign-backoffice-app/src/app/[locale]/[institution]/api-keys/create/_components/CreateApiKeyForm.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      institutionId: institution.id,
+    },
+    resolver: zodResolver(createApiKeyPayloadSchema),
+  });
+  const { setValue, watch, formState, handleSubmit } = methods;
+
+  const environment = watch("environment", "test");
+
+  useEffect(() => {
+    const random = Math.random().toString(32).substring(3, 7);
```
Comment (gunzip) [inline apps/io-sign-backoffice-app/src/app/[locale]/[institution]/api-keys/create/_components/CreateApiKeyForm.tsx:48]: can you add a comment?

File: apps/io-sign-backoffice-app/src/components/ThemeRegistry.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import { Options } from "@emotion/cache";
+import { useServerInsertedHTML } from "next/navigation";
+import { CacheProvider } from "@emotion/react";
+import { ThemeProvider } from "@mui/material/styles";
+import CssBaseline from "@mui/material/CssBaseline";
+import { theme } from "@pagopa/mui-italia";
+
+import * as ReactIs from "react-is";
+
+// This implementation is from emotion-js
```
Comment (gunzip) [inline apps/io-sign-backoffice-app/src/components/ThemeRegistry.tsx:15]: to do what? I'd add it as comment

File: apps/io-sign-backoffice-app/src/app/[locale]/[institution]/api-keys/[api-key]/_components/ApiKeyEditableFieldCard.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+import { ApiKeyContext, upsertApiKeyField } from "@/lib/api-keys/client";
+
+import FiscalCodeListInput from "@/components/FiscalCodeListInput";
+import IpAddressListInput from "@/components/IpAddressListInput";
+
+import { Props as EditableListProps } from "@/components/EditableList";
+
+type Props = {
+  field: "cidrs" | "testers";
```
[the remark this answers is not in the document]
Comment (lucacavallaro) [reply apps/io-sign-backoffice-app/src/app/[locale]/[institution]/api-keys/[api-key]/_components/ApiKeyEditableFieldCard.tsx:18]: it to the field of an `ApiKey` (the component name is `ApiKeyEditableField...` ) what could be a more appropriate name?

[the remark this answers is not in the document]
Comment (gunzip) [reply apps/io-sign-backoffice-app/src/app/[locale]/[institution]/api-keys/[api-key]/_components/ApiKeyEditableFieldCard.tsx:18]: what does represents an ApiKey field? Do you mean a form field?

**Comment by gunzip** [comment]
code lgtm, I'd add a readme with main infos

File: apps/io-sign-backoffice-app/src/app/[locale]/[institution]/api-keys/[api-key]/_components/ApiKeyEditableFieldCard.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (lucacavallaro) [reply apps/io-sign-backoffice-app/src/app/[locale]/[institution]/api-keys/[api-key]/_components/ApiKeyEditableFieldCard.tsx:18]: An "API Key field" it's a property of an API Key object

This component, given the field name (cidrs or testers, "field" property) shows its content (that is a list of string) and renders a form to change its value

**Pull request merged**